Allow bools and dates as options for random_choice's short syntax

prescod opened the pull request:
random_choice has a short-form key:probability syntax. Now that short-form syntax can be used for booleans and dates, not just strings.

boakley commented on snowfakery/parse_recipe_yaml.py:
```diff
@@ -146,14 +146,34 @@ def removeline_numbers(dct: Dict) -> Dict:
 
 
+def _coerce_to_string(val, context):
+    print(val, type(val))
```
Was this print statement intentionally left in?

prescod replied:
Oops, no, fixed in a0fb7ef.

prescod commented:
@boakley i fixed the issue you noticed, but it triggered a bug in `coverage`. The workaround is to improve coverage so I added an unrelated test to improve coverage.

boakley reviewed (commented):
Looks good to me.
